app/models/manageiq/providers/hawkular/middleware_manager/event_catcher_mixin.rb

```diff
@@ -0,0 +1,74 @@
+# This mixin allows us to share code easily between the event catcher runner and the event catcher spec. Otherwise
+# this code could just have been in the runner.
+module ManageIQ::Providers::Hawkular::MiddlewareManager::EventCatcherMixin
+  INCLUDE_EVENTS = {'Hawkular Deployment' => '.*'}.freeze # some Hash of included events
+
+  def event_monitor_handle
+    require 'middleware/events/middleware_event_monitor'
+    @event_monitor_handle ||= MiddlewareEventMonitor.new(@ems)
+  end
+
+  def reset_event_monitor_handle
+    @event_monitor_handle = nil
+  end
+
+  def stop_event_monitor
+    reset_event_monitor_handle
+  end
+
+  def monitor_events
+    event_monitor_running
+    event_monitor_handle.poll do |event|
+      if valid?(event)
+        $mw_log.info("Queuing Event #{event}") # TODO: maybe debug level
+        @queue.enq event
+      end
+    end
+  end
+
+  def process_event(event)
+    $mw_log.debug("Processing Event #{event}")
+
+    event_data = {
+      :timestamp => Time.at(event.ctime / 1000),
+      :category  => event.category,
+      :message   => event.context['Message'],
+      :reason    => event.text,
+      :resource  => event.context['CanonicalPath']
+    }
+
+    $mw_log.debug("Processing EventData #{event_data}")
+
+    # normalize event type to be valid (no invalid chars) and to easily match any we've made available for automation
+    event_type              = "#{event_data[:category]}.#{event_data[:reason]}"
+    event_type              = event_type.strip.gsub(/\s+/, "_").downcase
+    event_data[:event_type] = event_type
+
+    unless filtered?(event_data)
+      $mw_log.info "#{log_prefix} Adding ems event [#{event_data}]" # TODO: maybe debug level
+      event_hash = ManageIQ::Providers::Hawkular::MiddlewareManager::EventParser.event_to_hash(
+        event_data, @cfg[:ems_id])
+      EmsEvent.add_queue('add', @cfg[:ems_id], event_hash)
+    end
+  end
+
+  private
+
+  def valid?(event)
+    pattern = INCLUDE_EVENTS[event.category]
+    message = event.context['Message']
+    return true if pattern && message && message.match(pattern)
+
+    $mw_log.debug "#{log_prefix} Discarding event [#{event}]"
+    false
+  end
+
+  # Check for blacklisted event type
+  def filtered?(event_data)
+    if filtered_events.include?(event_data[:event_type])
```

Comment:
nitpicking on style:
If you move the logging to `process_event` you can a) shrink this method and move the logging concern to only `process_event`
[like this](https://github.com/durandom/manageiq/blob/master/gems/manageiq-providers-amazon/app/models/manageiq/providers/amazon/cloud_manager/event_catcher/runner.rb#L30)